src/storage/src/hummock/sstable/forward_sstable_iterator.rs

```diff
@@ -71,6 +71,12 @@ impl SSTableIterator {
             self.sst.value().id,
             idx,
         );
+
+        // When all data are in block cache, it is highly possible that this iterator will stay on a
+        // worker thread for a full time. Therefore, we use tokio's unstable API consume_budget to
+        // do cooperative scheduling.
+        tokio::task::consume_budget().await;
```

Comment:
is there a way we can achieve something similar without tokio unstable? Maybe we could yield when idx % 1000 == 0?